### ISSUE TITLE
Token creation issue with Okta / SAML / Kentor

### ISSUE THREAD

**Issue by dahlsailrunner** [IdentityServer/IdentityServer3#1956]
Hi -- 

I recently have been trying to get my instance of IdentityServer3 (v2) set up working with Okta via SAML.  With some help from Anders Abel I was able to get most of it working -- I now have a problem trying to sign the token I send back from Id Server to the client -- the one requesting authentication to start.

To get the SAML auth to work I had to add this line:

    Options.GlobalEnableSha256XmlSignatures();

This added line was based on the output of an exception thrown by the Kentor stuff (I can provide that if it would be helpful)

When I did that, though, I get the exception below from IdentityServer3 when trying to return the identity to the actual application.

Do you have any insight as to what my revised Id Server config should be when doing this?

Appreciate any insight - 
Erik

Message: 
 Exception:System.Security.Cryptography.CryptographicException
 Message: Invalid algorithm specified. 
 HResult : -2146893816
 Source:mscorlib
 StackTrace: at System.Security.Cryptography.CryptographicException.ThrowCryptographicException(Int32 hr) 

at System.Security.Cryptography.Utils.SignValue(SafeKeyHandle hKey, Int32 keyNumber, Int32 calgKey, Int32 calgHash, Byte[] hash, Int32 cbHash, ObjectHandleOnStack retSignature) 
 at System.Security.Cryptography.Utils.SignValue(SafeKeyHandle hKey, Int32 keyNumber, Int32 calgKey, Int32 calgHash, Byte[] hash) 
 at System.Security.Cryptography.RSACryptoServiceProvider.SignHash(Byte[] rgbHash, Int32 calgHash) 
 at System.IdentityModel.Tokens.JwtSecurityTokenHandler.CreateSignature(String inputString, SecurityKey key, String algorithm, SignatureProvider signatureProvider) in c:\workspace\WilsonForDotNet45Release\src\System.IdentityModel.Tokens.Jwt\JwtSecurityTokenHandler.cs:line 854 
 at System.IdentityModel.Tokens.JwtSecurityTokenHandler.WriteToken(SecurityToken token) in c:\workspace\WilsonForDotNet45Release\src\System.IdentityModel.Tokens.Jwt\JwtSecurityTokenHandler.cs:line 815 
 at IdentityServer3.Core.Services.Default.DefaultTokenSigningService.CreateJsonWebToken(Token token, SigningCredentials credentials) in c:\ballen\github\identity\IdSrv3\IdentityServer3\source\Core\Services\Default\DefaultTokenSigningService.cs:line 91 
 at IdentityServer3.Core.Services.Default.DefaultTokenSigningService.SignTokenAsync(Token token) in c:\ballen\github\identity\IdSrv3\IdentityServer3\source\Core\Services\Default\DefaultTokenSigningService.cs:line 55 
 at IdentityServer3.Core.Services.Default.DefaultTokenService.d__c.MoveNext() in c:\ballen\github\identity\IdSrv3\IdentityServer3\source\Core\Services\Default\DefaultTokenService.cs:line 193

**Comment by leastprivilege** [comment]
I remember vaguely that this has been discussed before. Search the archives.

**Comment by dahlsailrunner** [comment]
Sorry -- should have done that first.  Found this:
https://github.com/IdentityServer/IdentityServer3/issues/1903

I've added this to a question on the Kentor repo here:
https://github.com/KentorIT/authservices/issues/304 -- namely whether any change is coming on the Kentor side to accommodate this.

I'll probably go down the path of getting a new cert with the options in the last post in thread 1903 above though.  Thanks for the response.
